core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -141,6 +142,8 @@ public CloseableIterable<FileScanTask> planFiles() {
           doPlanFiles(),
           () -> {
             planningDuration.stop();
+            Map<String, String> metadata = Maps.newHashMap(context().options());
+            metadata.putAll(EnvironmentContext.get());
```

Comment:
I think having the scan options + the environment properties in the metadata is helpful

Reply:
it seems useful to know what `TableProperties.SPLIT_OPEN_FILE_COST` / `TableProperties.SPLIT_LOOKBACK` / `TableProperties.SPLIT_SIZE` were configured to for this particular scan

Reply:
Yes, but why mix the scan options and environment context together as "metadata"? I thought your comment above was to say that it is helpful to mix them?

Reply:
what I meant is that it's helpful to have both to be added as `metadata` to the scan report, since `metadata` is fairly generic

Reply:
How is it helpful? I see that metadata is generic, but I don't understand the value of mixing these together, versus having a separate context map and options map.

Reply:
I think we have the following options here:
1) define 2 separate maps, one for the context and one for the options
2) use `metadata` as a term and add both the env context info + the options to that map
3) use `metadata` as a term and just add the env context info

I think the decision should be based on the question: How much to we care about the clear distinction between "those are the options the scan has been configured with" and "this is some additional metadata the scan was running with"? 

Having two separate maps seems rather unnecessary to me, simply because there are currently exactly 3 scan options that can be configured. 

On the receiving side of the scan report I see the info from the env context and the scan options as **metadata** of the scan itself. The term **metadata** is fairly generic so that the information from the env and from the scan options make sense to be included. Additionally, we only have to deal with a single Map during ser/de.

That being said, I'm in favor of option 2)